e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts

```diff
@@ -0,0 +1,108 @@
+/// <reference types="Cypress" />
+
+import { Given, Then, When } from "@badeball/cypress-cucumber-preprocessor";
+import { sortingNameParamMap } from "@cypress-e2e/fixtures/constants";
+
+Given("The user is on the products page", () => {
+  cy.visit("http://localhost:3000/products");
+  cy.intercept("GET", /\/api\/v1\/products/).as("productsPageRequest");
+});
+
+Given("The user is on the first page of products", () => {
+  cy.visit("http://localhost:3000/products?page=1");
+  cy.intercept("GET", /\/api\/v1\/products/).as("productsPageRequest");
+});
+
+Given("The user is on the second page of products", () => {
+  cy.visit("http://localhost:3000/products?page=2");
+  cy.intercept("GET", /\/api\/v1\/products/).as("productsPageRequest");
+});
+
+When("The user looks at the pagination", () => {
+  cy.getById("pagination").should("be.visible");
+});
+
+When("The user looks at the product", () => {
+  cy.getById("product-card").first().should("be.visible");
+});
+
+When("Hovers product image", () => {
+  cy.getById("product-card-img").first().invoke("mouseover");
+  cy.getById("product-card-description").first().as("firstProductDescription");
+
+  cy.get("@firstProductDescription").should("be.hidden").then((el) => {
+    el.css("visibility", "visible");
+    el.css("opacity", "1");
+    el.css("transform", "translateY(0)");
+  });
+});
+
+Then("The user should see product description", () => {
+  cy.get("@firstProductDescription").should("be.visible");
+});
+
+When("The user looks at the products section", () => {
+  cy.getById("products-container").should("be.visible");
+});
+
+Then("The user should see {int} products", (productsCount) => {
+  cy.getById("product-card").should("have.length", productsCount);
+});
+
+When("The user looks at the sorting dropdown", () => {
+  cy.getById("products-dropdown").should("be.visible");
+});
+
+When("The user opens sorting dropdown", () => {
+  cy.getById("products-dropdown").click();
+});
+
+When("The user clicks {int} page", (page: number) => {
+  cy.intercept("GET", new RegExp("page=" + page)).as("productsPageRequest");
```

Comment:
use constant also there

Reply:
Thank you, I used constants here and in other places